crates/bevy_reflect/src/reflect.rs

```diff
@@ -238,15 +235,15 @@ impl dyn Reflect {
     /// otherwise.
     #[inline]
     pub fn is<T: Reflect>(&self) -> bool {
-        self.any().is::<T>()
+        Reflect::type_id(self) == TypeId::of::<T>()
```

Comment:
Just curious, why this instead of `self.any_ref().is::<T>()`?

Reply:
To me it's much cleaner and more intuitive to control `is` with `type_id` than `any_ref`; it should also be a tiny bit less work on each `downcast` call, compared to casting to &any and then to Box<any>.